src/actions/account.js

```diff
@@ -164,6 +164,29 @@ async function getAccounts(tokens, options) {
   }, Promise.resolve({}));
 }
 
+export const updatedLoggedinAddresses = tokens => async (dispatch, getState) => {
+  const { network: networkConfig, account, settings } = getState();
+  const newTokens = (Array.isArray(tokens)
+    ? tokens
+    : [tokens]
+  ).filter(key => settings.token.list[key]);
```

Comment:
Is it necessary to pass `tokens` into this action if you're reading `settings.token.list` inside anyway? Wouldn't be this enough? :

```suggestion
  Object.keys(settings.token.list).filter(key => settings.token.list[key]);
```

Currently, it feels like there's a part of the token filtering in the middleware and part in the action. IMO it's cleaner to put all token handling in one place, either here and don't pass any param, or in the middleware. and don't filter it here.

Reply:
Moved the filtering to the middleware, so the action is only called if the token was enabled, instead of triggering the action even if the token is being disabled.